src/main/java/Storage.java

```diff
@@ -0,0 +1,100 @@
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.util.Scanner;
+
+public class Storage { //deals with loading task and saving task in file
+
+  private String filePath;
+  private File file;
+
+  /**
+   * Creates a new Storage object. Storage is initialised with filepath passed from Bot.java.
+   * Filepath is assigned to this.filePath. Storage checks if there is a file named duke.txt (from
+   * the file path) that exists. If it does, the file is assigned to this.file
+   *
+   * @param filePath file path of duke.txt (or other file name as determined by creator)
+   */
+  public Storage(String filePath) {
+    this.filePath = filePath;
+
+    try {
+      File myFile = new File(filePath);
+      boolean exists = myFile.exists();
+      if (!exists) {
+        myFile.createNewFile();
+        this.file = myFile;
+        System.out.println("new duke.txt created");
+      } else if (exists) {
+        this.file = myFile;
+        System.out.println("file exists");
+      } else {
+        throw new IOException();
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Creates a new ArrayList<Listing>. If this.file contains a non-empty file, the lines are parsed
+   * and split using a Scanners to return a string array of size 3 containing the code, isDone,
+   * details and timedetail if applicable (or left null if none is available). The corresponding
+   * Listing is then created and added to the Arraylist which is then returned. If file is empty
+   * then an empty ArrayList is returned.
```

Comment:
I think you have to specify the return type of the method in your docs?

Reply:
Done